src/GenSequence.ts

```diff
@@ -27,6 +27,7 @@ export interface Sequence<T> extends IterableLike<T> {
     any(fnFilter: (t: T)=> boolean): boolean;
     first(fnFilter?: (t: T)=> boolean, defaultValue?: T): Maybe<T>;
     first(fnFilter: (t: T)=> boolean, defaultValue: T): T;
+    max(): Maybe<T>;
```

Comment:
It isn't really return time.

Reducers always end the sequence.
Mappers change the data being passed along.
Filters drop out unused values.

For example:
* `skip`, `filter`, and `take` are a form of *filter*
* `any`, `max` are reducers.
* `map`, `scan` are mappers.

`first` is a combination of a filter and a reducer.  But since it ends the sequence, I consider it to be a reducer.